src/hotspot/share/cds/filemap.cpp

```diff
@@ -2148,7 +2202,13 @@ bool FileMapInfo::map_heap_regions(int first, int max,  bool is_open_archive,
     si = space_at(i);
     size_t size = si->used();
     if (size > 0) {
-      HeapWord* start = (HeapWord*)start_address_as_decoded_from_archive(si);
+      HeapWord* start;
+      if (UseCompressedOops) {
+        start = (HeapWord*)start_address_as_decoded_from_archive(si);
+      } else {
+        assert(is_aligned(si->mapping_offset(), sizeof(HeapWord)), "must be");
+        start = (HeapWord*)(header()->heap_begin() + si->mapping_offset() + HeapShared::runtime_delta());
+      }
```

Comment:
This block can also be replaced with

```
HeapWord* start = (HeapWord*)heap_region_runtime_start_address(si);
```
The `is_aligned(si->mapping_offset()` check can be moved into `heap_region_runtime_start_address()`

Reply:
Done.